pkg/keyidentifier/keyidentifier_test.go

```diff
@@ -13,15 +14,24 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type spec struct {
```

Comment:
Why not just use `KeyInfo`?

Reply:
Mostly because I didn't want to try shoe-horn the expected data into the KeyInfo struct, especially if I wanted to add debugging info to the spec type.

Reply:
The json is a superset of the KeyInfo struct, right? KeyInfo just specifies what's parsed from it. Or do I have that backwards?

Reply:
Hmm, yea I suppose it is a superset of `KeyInfo`.. There are some things that are convenient to parse from the json to a go struct, and this allows for that nicely.

Do you think that there is significant benefit to using the same type for the test specification and the type that holds the key information? I personally don't see much value, but I could just be missing something.

Reply:
IME having less stuff is always less confusing. Fewer things to keep in my head. Fewer LoC. When they're different I immediately start trying to understand _why_ and whether the difference is important. 

I also think the resulting code is cleaner.

Compare:
```go
fromJson := readFileFromJson()

if includeField1 {
  require.Equal(t, fromJson.Field1, actual.Field1)
}
require.Equal(t, fromJson.Field2, actial.Field2)
require.Equal(t, fromJson.Field3, actial.Field3)
```

with
```go
fromJson := readFileFromJson()

if !includeField1 {
  fromJson.Field1 = ""
}

require.EqualValues(t,fromJson, actial.Field3)
```

Reply:
(updated example code)

Reply:
Hmm, ok continuing our slack conversation, the place that this becomes burdensome is the loss of the `Source` attribute. Knowing that a key is generated from `puttygen` vs `ssh-keygen` is really handy when testing the fingerprints, because `puttygen` only seems to do the old-style hex fingerprints. see https://github.com/kolide/launcher/pull/528/files#diff-2bb52927af21dd3478f1948bed54f864R96

Reply:
I suppose I could unmarshal the json a second time into a hash to get the source, but at that point we might as well just use the original type..

Reply:
That's a good reason not to just reuse. Bummer.

I'd still really avoid the duplication. Either with a nested type, or with the type embedding

Reply:
The spec type is now essentially a wrapper around the KeyInfo type w/an additional 'Source' attr